### PR TITLE
add convenience shortcut for `Path.toBeASymbolicLink` feature

### PR DESCRIPTION
**Pull request by fimbulwint**
For both infix and fluent API styles

______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/PathAssertionSamples.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        // Passes, as subject `link` is a symbolic link
+        expect(link).toBeASymbolicLink()
+    }
+
+    @Test
+    fun isNotASymbolicLink() {
+        val path = tempDir.resolve("somePath")
+
+        // Fails, as subject `path` is a not a symbolic link
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/PathAssertionSamples.kt:25]: It's correct but I think the example would be better if the file actually exists. I suggest you create a regular file at that place

File: apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/PathAssertionSamples.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        // Passes, as subject `link` is a symbolic link
+        expect(link) toBe aSymbolicLink
+    }
+
+    @Test
+    fun isNotASymbolicLink() {
+        val path = tempDir.resolve("somePath")
+
+        // Fails, as subject `path` is a not a symbolic link
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/samples/PathAssertionSamples.kt:26]: Same same as for fluent

File: logic/atrium-logic-jvm/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultPathAssertions.kt [hunk 3]
```diff
@@ -112,6 +112,15 @@ class DefaultPathAssertions : PathAssertions {
     override fun <T : Path> isDirectory(container: AssertionContainer<T>): Assertion =
         fileTypeAssertion(container, A_DIRECTORY) { it.isDirectory }
 
+    override fun <T : Path> toBeASymbolicLink(container: AssertionContainer<T>): Assertion =
+        changeSubjectToFileAttributes(container, NOFOLLOW_LINKS) { fileAttributesExpect ->
```
Comment (robstoll) [inline logic/atrium-logic-jvm/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultPathAssertions.kt:116]: I see that you copied the code of fileTypeAssertion due to the NOFOLLOW_LINKS.I suggest you add `linkOption: LinkOption? = null` to fileTypeAssertion instead

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathExpectationsSpec.kt [hunk 4]
```diff
@@ -794,6 +802,59 @@ abstract class PathExpectationsSpec(
...
+                contains(
+                    expectedMessage,
+                    "${WAS.getDefault()}: ${A_DIRECTORY.getDefault()}"
+                )
+            }
+        }
+
+        it("does not throw for a symbolic link") {
+            val target = tempFolder.resolve("target")
+            val link = Files.createSymbolicLink(tempFolder.resolve("link"), target)
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathExpectationsSpec.kt:853]:
```suggestion
            val link = tempFolder.newSymbolicLink("link", target)
```

**Review by robstoll — CHANGES_REQUESTED** [review]
@fimbulwint thanks for your first contribution. A few improvements and we are good to merge it

**Review by robstoll — APPROVED** [review]

**Comment by robstoll** [comment]
@fimbulwint thanks for your first contribution :slightly_smiling_face: :+1: 

We could use your help for other issues, and don't forget to :star: Atrium:
https://github.com/robstoll/atrium/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22

**Comment by fimbulwint** [comment]
@robstoll happy to contribute :slightly_smiling_face:

I'll take a look at some other issue when I get the chance.

**Pull request merged**